### PR TITLE
Add comments to SD

### PR DESCRIPTION
**Pull request by python273**
in `master` fails with
```python
TypeError: Dummy() takes no arguments
```
and after adding `__init__`
```python
UnpicklingError: state is not a dictionary
```

### CODE REVIEW

**Comment by geohot** [comment]
I can't repro. I think you have different weights.

**Comment by python273** [comment]
hmm, something in my python env triggers it 🤔 . Seems to be working in a clean env

**Comment by python273** [comment]
seems like if you have `pytorch-lightning` installed, while unpickling weights, it gets imported and it breaks something

`pytorch_lightning.callbacks.model_checkpoint ModelCheckpoint`

https://github.com/geohot/tinygrad/blob/58d3824cbe93fb50ca350f6f40fcd2646a487c59/extra/utils.py#L60-L62

File: extra/utils.py [hunk 1]
```diff
@@ -74,8 +74,9 @@ def find_class(self, module, name):
         elif name == "_rebuild_parameter":
           return HackParameter
       else:
+        if module.startswith('pytorch_lightning'): return Dummy
         try:
-          return pickle.Unpickler.find_class(self, module, name)
+          return super().find_class(self, module, name)
```
Comment (python273) [inline extra/utils.py:79]: `pytorch_lightning` does monkeypatching on Unpickler class, and it breaks `MyPickle`
https://github.com/Lightning-AI/lightning/blob/35d15621c7d007fa7418958b9eef120e14abc1a9/src/lightning/pytorch/_graveyard/legacy_import_unpickler.py#L46
```
Traceback (most recent call last):
  File "/home/kirill/3rdworkspace/tinygrad/extra/utils.py", line 92, in find_class
    cla = pickle.Unpickler.find_class(self, module, name)
  File "/home/kirill/miniconda3/envs/nnexp/lib/python3.10/site-packages/pytorch_lightning/_graveyard/legacy_import_unpickler.py", line 25, in find_class
    return super().find_class(new_module, name)
TypeError: super(type, obj): obj must be an instance or subtype of type
```

File: extra/utils.py [hunk 2]
```diff
@@ -74,8 +74,9 @@ def find_class(self, module, name):
         elif name == "_rebuild_parameter":
           return HackParameter
       else:
+        if module.startswith('pytorch_lightning'): return Dummy
```
Comment (python273) [inline extra/utils.py:77]: also `pytorch_lightning` enables this fancy traceback printer, so maybe better to skip importing it completely?

https://rich.readthedocs.io/en/stable/traceback.html

**Comment by geohot** [comment]
cool, merged

**Pull request merged**